src/endpoint/scheduler.rs

```diff
@@ -370,6 +374,36 @@ impl JobHandle {
     }
 }
 
+impl Drop for JobHandle {
+    fn drop(&mut self) {
+        debug!("Cleaning up JobHandle");
+        if self.container_id.is_some() {
+            debug!("Container was already started");
+            let docker = self.endpoint.docker().clone();
+            let container_id = self.container_id.take().unwrap();
+
+            tokio::spawn(async move {
+                let container = docker.containers().get(&container_id);
+                let container_info = container.inspect().await.unwrap();
+
+                if container_info.state.running {
+                    debug!("Container is still running, cleaning up...");
+                    match container.kill(None).await {
+                        Ok(_) => debug!("Stopped container with id: {}", container_id),
```

Comment:
The text doesn't match the action - there are both `stop` and `kill` methods so we should use "Killed" here (IMO "Terminated" sounds better but I'd probably stick with the naming of the crate and Docker API here). The same applies for the error handling.

Reply:
I was stopping the containers before, forgot to update the debug message. Thanks

Reply:
In that case it would probably also be nice to document why we're killing the containers instead of stopping them. I was already wondering about that - unfortunately the Docker API documentation didn't state the differences. I assume that killing is a more reliable way but I didn't take a deeper look.